### PR TITLE
Separate our additions from the original crate

### PR DESCRIPTION
**Pull request by moCello**
Resolves: #109

### CODE REVIEW

File: src/fp2.rs [hunk 1]
```diff
@@ -422,545 +332,631 @@ impl Fp2 {
...
-            .lexicographically_largest()
-        ));
+
+            const FIELDS: &[&str] = &["c0", "c1"];
+            deserializer.deserialize_struct("Fp2", FIELDS, Fp2Visitor)
+        }
     }
 
     #[test]
-    #[cfg(feature = "serde_req")]
```
Comment (ureeves) [inline src/fp2.rs:963]: Does this file only compile with `serde_req`? If not this needs to stay.

File: src/lib.rs [hunk 2]
```diff
@@ -43,14 +43,17 @@ pub mod notes {
     pub mod serialization;
 }
 
-mod choice;
+mod dusk;
+use dusk::choice;
```
Comment (ureeves) [inline src/lib.rs:47]: Unused?

File: src/notes/design.rs [hunk 3]
```diff
@@ -18,45 +18,45 @@
 //!
 //! This can be derived using the following sage script:
 //!
-//! ```python
+//! ```text
```
Comment (ureeves) [inline src/notes/design.rs:21]: Wait, is this not python?

File: src/pairings.rs [hunk 4]
```diff
@@ -62,12 +57,12 @@ impl MillerLoopResult {
...
-            let mut z3 = f.c0.c2.clone();
-            let mut z2 = f.c1.c0.clone();
-            let mut z1 = f.c1.c1.clone();
-            let mut z5 = f.c1.c2.clone();
+            let mut z0 = f.c0.c0;
+            let mut z4 = f.c0.c1;
+            let mut z3 = f.c0.c2;
+            let mut z2 = f.c1.c0;
+            let mut z1 = f.c1.c1;
+            let mut z5 = f.c1.c2;
```
Comment (ureeves) [inline src/pairings.rs:65]: Nice!

File: src/scalar.rs [hunk 5]
```diff
@@ -894,449 +540,807 @@ impl<'a> From<&'a Scalar> for [u8; Scalar::SIZE] {
...
     }
+}
 
-    #[cfg(test)]
-    const LARGEST: Scalar = Scalar([
-        0xffffffff00000000,
+#[test]
+fn test_invert_is_pow() {
+    let q_minus_2 = [
+        0xfffffffeffffffff,
```
Comment (ureeves) [inline src/scalar.rs:926]: Why are these numbers not separated by `_`?

**Review by ureeves — CHANGES_REQUESTED** [review]
Looks good, but I'd like us to pause before some changes.

File: src/fp.rs [hunk 6]
```diff
@@ -676,321 +612,390 @@ impl Fp {
...
-        ]);
-        let b = Fp([
-            0x69ecd7040952148f,
-            0x985ccc2022190f55,
-            0xe19bba36a9ad2f41,
-            0x19bb16c95219dbd8,
-            0x14dcacfdfb478693,
-            0x115ff58afff9a8e1,
-        ]);
+    #[cfg(feature = "serde_req")]
```
[the remark this answers is not in the document]
Comment (moCello) [reply src/fp.rs:943]: I think that's a good idea but I would prefer doing that in a different issue

File: src/fp2.rs (hunk 1 above)
Comment (moCello) [reply src/fp2.rs:963]: The whole `dusk` module only compiles with "serde_req" so I moved the feature flag to line 868

File: src/notes/design.rs (hunk 3 above)
Comment (moCello) [reply src/notes/design.rs:21]: It says `text` in the original file

File: src/scalar.rs (hunk 5 above)
Comment (moCello) [reply src/scalar.rs:926]: Good catch, they should! Will fix

**Pull request merged**